Core/include/Acts/Utilities/Grid.hpp

```diff
@@ -465,20 +477,61 @@ class Grid final {
     return current_size;
   }
 
+  /// @brief Convenience function to convert the type of the grid
+  /// to hold another object type.
+  ///
+  /// @tparam U the new grid value type
+  ///
+  /// @return a new grid with the same axes and a different value type
+  template <typename U>
+  Grid<U, Axes...> convertType() const {
+    Grid<U, Axes...> cGrid(m_axes);
+    return cGrid;
+  }
+
+  /// @brief Convenience function to convert the type of the grid
+  /// to hold another object type.
+  ///
+  /// @tparam U the new grid value type
+  /// @tparam CT the converter type
+  ///
+  /// @return a new grid with the same axes and a different value type
+  template <typename U, typename CT>
+  Grid<U, Axes...> convertGrid() const {
+    Grid<U, Axes...> cGrid(m_axes);
+    // Loop through the values and convert them
+    for (std::size_t i = 0; i < size(); i++) {
+      cGrid.at(i) = CT::convert(at(i));
```

Comment:
this relies on `CT` to have `convert`. Why not using `operator()` instead?

Is there a way of doing some kind of static assert (without C++20 I mean) to check `CT::convert` exists?

Reply:
Good point, will change.

Reply:
I actually changed quite a bit, if we are given a `converter_type` as template object, it also should define the return type, so there's no mismatch.

Reply:
The updated syntax is 

 - `convertGrid<Converter>(Converter& c)`

With the converter as a visitor.